pageserver/src/tenant.rs

```diff
@@ -422,13 +422,46 @@ impl Tenant {
             init_order,
             CreateTimelineCause::Load,
         )?;
-        let new_disk_consistent_lsn = timeline.get_disk_consistent_lsn();
+        let disk_consistent_lsn = timeline.get_disk_consistent_lsn();
+        assert_eq!(
+            disk_consistent_lsn,
+            up_to_date_metadata.disk_consistent_lsn(),
+            "these are used interchangeably; need to move layer map loading before Timeline creation to avoid drifting"
```

Comment:
```suggestion
            "timeline must not yet advance"
```

Reply:
As in, when I started, I think there were two, but they were basically the same. Maybe I could just use the `up_to_date_metadata.disk_consistent_lsn()` actually.

Reply:
Well, unsure, just changing the message.

Reply:
Unsure because otherwise we use timeline getters.

It'd be cool if the latest metadata was now moved to timeline, we couldn't use it anymore.

Reply:
0de2cf149